src/compo/modis_aod2ioda.py

```diff
@@ -14,6 +14,7 @@
 from datetime import datetime, date, timedelta
 import os
 from pathlib import Path
+from pyhdf.SD import SD, SDC
```

Comment:
Why switch to pyhdf API? What problem is this solving?

Reply:
Because I had to.  The input file is hdf4, not hdf5/netcdf, so I had to switch.  This is basically coming from a develop branch change that someone made - so I just followed it thru into the sprint branch so it functions again.

Reply:
also found this strange was this something from the aerosol group?  but if `pyhdf` is supported in the stack would potentially get this through and then have the aerosol group review and hopefully consolidate and minimize the number of libraries used

Reply:
@gthompsnJCSDA thanks for the explanation. pyhdf is supported/included in spack-stack. It would be nice in the future to get providers updated from hdf4 to netcdf/hdf5, but that is certainly outside the scope of this PR.